loki/transform/transform_hoist_variables.py

```diff
@@ -356,6 +364,30 @@ def kernel_call_argument_remapping(self, routine, call, variables):
         new_args = tuple(v.clone(dimensions=None) for v in variables)
         return call.clone(arguments=call.arguments + new_args)
 
+    def kernel_inline_call_argument_remapping(self, routine, call, variables):
+        """
+        Append hoisted temporaries to inline function call arguments.
+
+        Parameters
+        ----------
+        routine : :any:`Subroutine`
+            The subroutine to add the variable declaration to.
+        call : :any:`InlineCall`
+            ProcedureSymbol to which hoisted variables will be added.
+        variables : tuple of :any:`Variable`
+            The tuple of variables to be declared.
+        """
+
+        if self.as_kwarguments:
+            new_args = dict((a.name, v.clone(dimensions=None)) for (a, v) in variables)
+            _call_clone = call.clone(kwparameters=call.kw_parameters)
+            _call_clone.kw_parameters.update(new_args)
+            vmap = {call: _call_clone}
+            routine.body = SubstituteExpressions(vmap).visit(routine.body)
+        else:
+            new_args = tuple(v.clone(dimensions=None) for v in variables)
+            vmap = {call: call.clone(parameters=call.parameters + new_args)}
+            routine.body = SubstituteExpressions(vmap).visit(routine.body)
```

Comment:
You could move `routine.body = SubstituteExpressions(vmap).visit(routine.body)` outside the `if`/`else` statement ...

Reply:
Ah yes, good spot, thanks!